### PR TITLE
Add model for bgp global configuration for IOSXR

### PR DESCRIPTION
**Pull request by ashwini-mhatre**

### CODE REVIEW

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml [hunk 1]
```diff
@@ -0,0 +1,826 @@
+module: iosxr_bgp_global
+short_description: Manages BGP global resource module.
+description:
+- This module configures and manages the attributes of BGP global on Cisco IOS-XR platforms.
+version_added: 1.0.0
```
Comment (NilashishC) [inline models/iosxr/bgp_global/iosxr_bgp_global.yaml:5]: This should be the next minor release version of IOS-XR.

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml [hunk 2]
```diff
@@ -0,0 +1,826 @@
...
+author: Ashwini Mhatre (amhatre)
+notes:
+- Tested against Cisco IOS-XR 6.1.3.
+- This module works with connection C(network_cli).
+options:
+    config:
+      description: A list of configurations for BGP global.
+      type: dict
+      suboptions:
+        as_number:
```
Comment (NilashishC) [inline models/iosxr/bgp_global/iosxr_bgp_global.yaml:15]: IMO, we should be standardizing the keys based on OpenConfig.
https://github.com/openconfig/public/blob/master/release/models/bgp/openconfig-bgp-global.yang#L103

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml [hunk 3]
```diff
@@ -0,0 +1,826 @@
...
+          description: Autonomous system number of the router.
+          type: str
+          required: true
+        bfd_params: &bfd_params
```
Comment (NilashishC) [inline models/iosxr/bgp_global/iosxr_bgp_global.yaml:19]: `_params` suffix can be removed.

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml [hunk 4]
```diff
@@ -0,0 +1,826 @@
...
+          description: Configure BFD parameters.
+          type: dict
+          suboptions:
+            minimum_interval: &min_interval
+              description: Specifies the BFD session's minimum-interval value for the neighbor.
+              type: int
+            multiplier: &multiplier
+              description: Specifies the BFD session's multiplier value for the neighbor.
+              type: int
+        bgp_params: &bgp_params
```
Comment (NilashishC) [inline models/iosxr/bgp_global/iosxr_bgp_global.yaml:29]: Same as above.

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml [hunk 5]
```diff
@@ -0,0 +1,826 @@
...
+          description: Control distribution of default information.
+          type: dict
+          suboptions:
+            originate:
+              description: Distribute a default route
+              type: bool
+        default_metric: &default_metric
+          description: Default metric. Example-<1-4294967295>.
+          type: in
+        graceful_maintenanc:
```
Comment (NilashishC) [inline models/iosxr/bgp_global/iosxr_bgp_global.yaml:243]:
```suggestion
        graceful_maintenance:
```

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml [hunk 6]
```diff
@@ -0,0 +1,826 @@
...
+                Attributes for Graceful Maintenance. This will cause neighbors to de-prefer routes from this router and
+                choose alternates. This allows the router to be brought in or out of service gracefully.
+              type: dict
+              suboptions:
+                activate:
+                  description: Routes will be announced with the graceful maintenance attributes while activated either here or under router
+                    bgp configuration.
+                  type: dict
+                  suboptions:
+                    inheritance-disable:
```
Comment (NilashishC) [inline models/iosxr/bgp_global/iosxr_bgp_global.yaml:401]: `-` is not a valid character in parameter name.

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml [hunk 7]
```diff
@@ -0,0 +1,826 @@
...
+                      description: Prevent activate from being inherited from the parent.
+                      type: bool
+                as_prepends:
+                  description: Number of times to prepend the local AS number to the
+                    AS path of routes. Default=0
+                  type: dict
+                  suboptions:
+                    inheritance-disable:
```
Comment (NilashishC) [inline models/iosxr/bgp_global/iosxr_bgp_global.yaml:409]: Same as above.

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml [hunk 8]
```diff
@@ -0,0 +1,826 @@
...
+                restart_time: *restart_time
+                stalepath_time: *stalepath_time
+                disable:
+                  description: Enable graceful-restart.
+                  type: bool
+            ignore_connected_check: &ignore_connected_chk
+              description: Bypass the directly connected nexthop check for single-hop eBGP peering
+              type: dict
+              suboptions:
+                inheritance-disable:
```
Comment (NilashishC) [inline models/iosxr/bgp_global/iosxr_bgp_global.yaml:435]: Same as above.

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml [hunk 9]
```diff
@@ -0,0 +1,826 @@
...
+        transforms it into Ansible structured data as per the resource module's argspec
+        and the value is then returned in the I(parsed) key within the result.
+      type: str
+    state:
+      description:
+      - The state the configuration should be left in.
+      type: str
+      choices: [deleted, merged, overridden, replaced, gathered, rendered, parsed]
+      default: merged
+EXAMPLES:
```
Comment (NilashishC) [inline models/iosxr/bgp_global/iosxr_bgp_global.yaml:820]: The updated builder no longer handles `EXAMPLES`.

**Review by NilashishC — CHANGES_REQUESTED** [review]

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml (hunk 1 above)
Comment (ashwini-mhatre) [reply models/iosxr/bgp_global/iosxr_bgp_global.yaml]: Ack, made changes

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml (hunk 2 above)
Comment (ashwini-mhatre) [reply models/iosxr/bgp_global/iosxr_bgp_global.yaml:15]: Ack, made changes

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml (hunk 3 above)
Comment (ashwini-mhatre) [reply models/iosxr/bgp_global/iosxr_bgp_global.yaml]: Ack, made changes

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml (hunk 4 above)
Comment (ashwini-mhatre) [reply models/iosxr/bgp_global/iosxr_bgp_global.yaml]: Ack, made changes

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml (hunk 5 above)
Comment (ashwini-mhatre) [reply models/iosxr/bgp_global/iosxr_bgp_global.yaml]: Ack, made changes

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml (hunk 6 above)
Comment (ashwini-mhatre) [reply models/iosxr/bgp_global/iosxr_bgp_global.yaml]: Ack, made changes

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml (hunk 9 above)
Comment (ashwini-mhatre) [reply models/iosxr/bgp_global/iosxr_bgp_global.yaml]: Ack, made changes

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml (hunk 8 above)
Comment (ashwini-mhatre) [reply models/iosxr/bgp_global/iosxr_bgp_global.yaml]: Ack, made changes

File: models/iosxr/bgp_global/iosxr_bgp_global.yaml (hunk 7 above)
Comment (ashwini-mhatre) [reply models/iosxr/bgp_global/iosxr_bgp_global.yaml]: Ack, made changes

**Review by NilashishC — APPROVED** [review]

**Pull request merged**